tests/test_telegramobject.py

```diff
@@ -145,3 +147,33 @@ def test_subscription(self):
         assert message['from_user'] is user
         with pytest.raises(KeyError, match="Message don't have an attribute called `no_key`"):
             message['no_key']
+
+    def test_deepcopy_and_pickle(self, bot):
+        chat = Chat(2, Chat.PRIVATE)
+        user = User(3, 'first_name', False)
+        date = datetime.datetime.now()
+        photo = PhotoSize('file_id', 'unique', 21, 21, bot=bot)
+        msg = Message(1, date, chat, from_user=user, text='foobar', bot=bot, photo=[photo])
+
+        # Test pickling of TGObjects, we choose Message since it's contains the most subclasses.
+        assert msg.get_bot()
+        unpickled = pickle.loads(pickle.dumps(msg))
+
+        with pytest.raises(RuntimeError):
+            unpickled.get_bot()  # There should be no bot when we pickle TGObjects
+
+        assert unpickled.chat == chat
+        assert unpickled.from_user == user
+        assert unpickled.date == date
+        assert unpickled.photo[0] == photo
+
+        # Now lets test deepcopying
```

Comment:
please test deepcopy in a separate test

Reply:
Well it does share the code, agh guess I'll make it into two